homepage/homepage.go

```diff
@@ -34,100 +30,27 @@ type MainFigure struct {
-			for _, uri := range figure.uris {
-				zebResponse, err := zcli.GetTimeseriesMainFigure(ctx, userAccessToken, collectionID, lang, uri)
-				if err != nil {
-					log.Event(ctx, "error getting timeseries data", log.ERROR, log.Error(err), log.Data{"timeseries-data": uri})
-					mappedErrorFigure := &model.MainFigure{ID: id}
-					responses <- mappedErrorFigure
-					return
-				}
-				zebResponses = append(zebResponses, zebResponse)
-			}
-			trendInfo := getTrendInfo(ctx, userAccessToken, collectionID, lang, zcli, figure)
-			latestMainFigure := getLatestTimeSeriesData(ctx, zebResponses)
-			mappedMainFigure := mapper.MainFigure(ctx, id, figure.datePeriod, figure.differenceInterval, latestMainFigure, trendInfo)
-			responses <- mappedMainFigure
-			return
-		}(ctx, zcli, id, figure)
-	}
-	wg.Wait()
-	close(responses)
-
-	for response := range responses {
-		log.Event(ctx, "the response of this request was", log.ERROR, log.Data{"response": response})
-		mappedMainFigures[response.ID] = response
-	}
-
-	weekAgoTime := time.Now().AddDate(0, 0, -7)
-	dateFromDay := weekAgoTime.Format("02")
-	dateFromMonth := weekAgoTime.Format("01")
-	dateFromYear := weekAgoTime.Format("2006")
-	releaseCalResp, err := bcli.GetReleaseCalendar(ctx, userAccessToken, dateFromDay, dateFromMonth, dateFromYear)
+	homepageHTML, err := homepageClient.GetHomePage(ctx, userAccessToken, collectionID, lang)
 	if err != nil {
-		log.Event(ctx, "error failed to get release calendar data from babbage ", log.ERROR, log.Error(err))
-	}
-	releaseCalModelData := mapper.ReleaseCalendar(releaseCalResp)
-
-	// Get homepage data from Zebedee
-	homepageContent, err := zcli.GetHomepageContent(ctx, userAccessToken, collectionID, lang, HomepagePath)
-	if err != nil {
-		log.Event(ctx, "error getting homepage data from client", log.ERROR, log.Error(err), log.Data{"content-path": HomepagePath})
-	}
-
-	var mappedFeaturedContent []model.Feature
-	if len(homepageContent.FeaturedContent) > 0 {
-		imageObjects := map[string]image.ImageDownload{}
-		for _, fc := range homepageContent.FeaturedContent {
-			if fc.ImageID != "" {
-				image, err := icli.GetDownloadVariant(ctx, userAccessToken, "", "", fc.ImageID, ImageVariant)
-				if err != nil {
-					log.Event(ctx, "error getting image download variant", log.ERROR, log.Error(err), log.Data{"featured-content-entry": fc.Title})
-				}
-				imageObjects[fc.ImageID] = image
-			}
-		}
-		mappedFeaturedContent = mapper.FeaturedContent(homepageContent, imageObjects)
-	}
-
-	m := mapper.Homepage(lang, mappedMainFigures, releaseCalModelData, &mappedFeaturedContent, homepageContent.ServiceMessage)
-
-	b, err := json.Marshal(m)
-	if err != nil {
-		log.Event(ctx, "error marshalling body data to json", log.ERROR, log.Error(err))
-		http.Error(w, "error marshalling body data to json", http.StatusInternalServerError)
+		log.Event(ctx, "HOMEPAGE_RESPONSE_FAILED. failed to write response for homepage", log.ERROR, log.Error(err))
```

Comment:
I don't think this error is as descriptive as it could be.. If the operation that's failed is getting the page HTML it would be more helpful to have 'failed to get page html' as the description than just 'failed to write response'